stellargraph/mapper/sequences.py

```diff
@@ -273,20 +272,12 @@ def __init__(self, sample_function, batch_size, walker, shuffle=True):
                 "({}) UnsupervisedSampler is required.".format(type(self).__name__)
             )
 
-        self.lock = threading.Lock()
         self.batch_size = batch_size
         self.walker = walker
-
-        # an estimate of the  upper bound on how many samples are generated in each epoch
-        self.data_size = (
-            2
-            * len(self.walker.nodes)
-            * self.walker.length
-            * self.walker.number_of_walks
-        )
-
-        # the generator method from the sampler with the batch-size from the link generator method
-        self._gen = self.walker.generator(self.batch_size)
+        self.shuffle = shuffle
```

Comment:
Was the shuffle argument previously being ignored?

Reply:
yes, because there was nothing to shuffle, since each batch is computed on the fly

Reply:
Couldn't (shouldn't?) it still shuffle the order of the head IDs? (This is probably not a question that's relevant to this PR.)

Reply:
I think it was always just shuffling without end, there wasn't really a concrete idea of an "epoch" the way `UnsupervisedSampler.generator` was implemented - just an infinite loop that shuffles things when it runs out of nodes to go through and it wasn't guaranteed that an actual training epoch went through an "epoch" in the sampler.